addons/point_of_sale/static/src/js/custom_hooks.js

```diff
@@ -172,5 +172,39 @@ odoo.define('point_of_sale.custom_hooks', function (require) {
         };
     }
 
-    return { useErrorHandlers, useAutoFocusToLast, onChangeOrder, useBarcodeReader, useAsyncLockedMethod };
+    function useValidateCashInput(inputRef, startingValue) {
+        const cashInput = useRef(inputRef);
+        const current = Component.current;
+        const decimalPoint = current.env._t.database.parameters.decimal_point;
+        const thousandsSep = current.env._t.database.parameters.thousands_sep;
+        // Replace the thousands separator and decimal point with regex-escaped versions
+        const escapedThousandsSep = _.str.escapeRegExp(thousandsSep);
+        const scapedDecimalPoint = _.str.escapeRegExp(decimalPoint);
```

Comment:
We should use the one from the `web` addon. Look for:

`import { escapeRegExp } from "@web/core/utils/strings";`

Reply:
I added escapeRegExp from web addon, and also fixed it for the cash move popup.